internal/providers/terraform/terragrunt_hcl_provider.go

```diff
@@ -43,6 +43,10 @@ import (
 // concurrency safe downloading.
 var terragruntSourceLock = infSync.KeyMutex{}
 
+// terragruntDownloadedDirs is used to ensure sources are only downloaded once. Use terragruntSourceLock
+// for concurrency safety.
+var terragruntDownloadedDirs = map[string]bool{}
```

Comment:
```suggestion
var terragruntDownloadedDirs = map[string]bool{}
```

Maps are not concurrent safe this will need to be `sync.Map`